src/cpp/server/residency/platform/durable_file_adapter_macos.cpp

```
@@ -0,0 +1,977 @@
+        return failure;
+    }
+
+    int directory_fd_;
+    int lock_fd_ = -1;
+    bool directory_lock_held_ = false;
+};
+
+} // namespace
+
+std::unique_ptr<DurableFileAdapter>
+make_platform_durable_file_adapter(const std::filesystem::path &directory) {
```

Cppcheck (2.21.0)

[style] The function 'make_platform_durable_file_adapter' is never used.

(unusedFunction)